src/plugins/discover/public/application/components/sidebar/discover_field_details.tsx

```diff
@@ -39,6 +39,18 @@ export function DiscoverFieldDetails({
   onAddFilter,
 }: DiscoverFieldDetailsProps) {
   const warnings = getWarnings(field);
+  const [showVisualizeLink, setShowVisualizeLink] = useState<boolean | undefined>(false);
```

Comment:
:art: Could we just make `isFieldVisualizable` return `false` in case no index pattern is passed in and just get rid of the `undefined` here?